src/f_wipe.c

```diff
@@ -89,16 +90,16 @@ static int wipe_initMelt(int width, int height, int ticks)
 
   // setup initial column positions (y<0 => not ready to scroll yet)
   y = (int *) Z_Malloc(width*sizeof(int), PU_STATIC, 0);
-  y[0] = -(M_Random()%16);
+  y[0] = -(M_Random()%16) * hires_size;
   for (i=1;i<width;i++)
     {
-      int r = (M_Random()%3) - 1;
+      int r = ((M_Random()%3) - 1) * hires_size;
       y[i] = y[i-1] + r;
       if (y[i] > 0)
         y[i] = 0;
       else
-        if (y[i] == -16)
-          y[i] = -15;
+        if (y[i] == -16 * hires_size)
```

Comment:
I feel like I should mention that this check is prone to fail if the melt initialization is done taking full advantage of the resolution (e.g. `(M_Random() % (3 * hires_size)) - 1` instead of the current code). If that's ever the case, the check should probably be changed to `<=` instead of `==`:

Reply:
I am not exactly sure what you mean here? Why would we want to change the modulo divisor if we already scale the result with `hires_size`?

Reply:
What I said regarding the check isn't relevant to the current implementation in this PR. As I said, I just felt like mentioning it in case we were to take an alternative approach, which I'll explain below. The current approach works well enough in my opinion, though, so feel free to move along and ignore the explanation.

`((M_Random()%3) - 1)` yields integers in the `[-1, 1]` range. Multiply that by `hires == 1`, and we get `-2`, `0` and `2` as the only possible values.

The example I gave before is a bit incorrect; I actually meant `r = (M_Random() % (3 * hires_size)) - (1 * hires_size)`.
If we were to calculate `r` in that manner, we would instead get any integers in the `[-2, 3]` range, which could maybe lead to a more "natural" melt. I did try doing this (changing the rest of relevant calculations, of course), but I didn't like the results. Maybe I did something wrong.

If we were to take that alternative approach, `y[i]` _could_ go below `-16 * hires_size`, hence "skipping" the check. Only in that case would we have to change the check to `<=`.

Reply:
I see, thanks for the explanation! 👍